pkg/chunk/dynamodb_table_client.go

```diff
@@ -157,10 +193,70 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 			return err
 		})
 	})
+
+	if d.ApplicationAutoScaling != nil {
+		err = d.backoffAndRetry(ctx, func(ctx context.Context) error {
+			return instrument.TimeRequestHistogram(ctx, "ApplicationAutoScaling.DescribeScalableTargetsWithContext", applicationAutoScalingRequestDuration, func(ctx context.Context) error {
+				out, err := d.ApplicationAutoScaling.DescribeScalableTargetsWithContext(ctx, &applicationautoscaling.DescribeScalableTargetsInput{
+					ResourceIds:       []*string{aws.String("table/" + desc.Name)},
+					ScalableDimension: aws.String("dynamodb:table:WriteCapacityUnits"),
+					ServiceNamespace:  aws.String("dynamodb"),
+				})
+				switch l := len(out.ScalableTargets); l {
+				case 0:
+					return err
+				case 1:
+					desc.WriteScaleEnabled = true
+					desc.WriteScaleRoleARN = *out.ScalableTargets[0].RoleARN
+					desc.WriteScaleMinCapacity = *out.ScalableTargets[0].MinCapacity
+					desc.WriteScaleMaxCapacity = *out.ScalableTargets[0].MaxCapacity
+					return err
+				default:
+					return fmt.Errorf("more than one scalable target found for DynamoDB table")
+				}
+			})
+		})
+
+		err = d.backoffAndRetry(ctx, func(ctx context.Context) error {
+			return instrument.TimeRequestHistogram(ctx, "ApplicationAutoScaling.DescribeScalingPoliciesWithContext", applicationAutoScalingRequestDuration, func(ctx context.Context) error {
+				out, err := d.ApplicationAutoScaling.DescribeScalingPoliciesWithContext(ctx, &applicationautoscaling.DescribeScalingPoliciesInput{
+					PolicyNames:       []*string{aws.String(autoScalingPolicyNamePrefix + desc.Name)},
+					ResourceId:        aws.String("table/" + desc.Name),
+					ScalableDimension: aws.String("dynamodb:table:WriteCapacityUnits"),
+					ServiceNamespace:  aws.String("dynamodb"),
+				})
+				switch l := len(out.ScalingPolicies); l {
+				case 0:
+					return err
+				case 1:
+					desc.WriteScaleInCooldown = *out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration.ScaleInCooldown
+					desc.WriteScaleOutCooldown = *out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration.ScaleOutCooldown
+					desc.WriteScaleTargetValue = *out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration.TargetValue
+					return err
+				default:
+					return fmt.Errorf("more than one scaling policy found for DynamoDB table")
+				}
+			})
+		})
+	}
 	return
 }
 
 func (d dynamoTableClient) UpdateTable(ctx context.Context, current, expected TableDesc) error {
+	disableAutoScaling := current.WriteScaleEnabled && !expected.WriteScaleEnabled
+	enableAutoScaling := !current.WriteScaleEnabled && expected.WriteScaleEnabled
+	updateAutoScaling := current.WriteScaleEnabled && expected.WriteScaleEnabled && !current.AutoScalingEquals(expected)
+	if disableAutoScaling {
+		err := d.disableAutoScaling(ctx, expected)
+		if err != nil {
+			return err
+		}
+	} else if enableAutoScaling || updateAutoScaling {
+		err := d.enableAutoScaling(ctx, expected)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Not sure, but I think I'd write it like this:

```go
var err error
if !current.WriteScaleEnabled {
	if expected.WriteScaleEnabled {
		err = d.enableAutoScaling(ctx, expected)
	}
} else {
	if !expected.WriteScaleEnabled {
		err = d.disableAutoScaling(ctx, expected)
	} else if !current.AutoScalingEquals(expected) {
		err = d.enableAutoScaling(ctx, expected)
	}
}
if err != nil {
	return err
}
```

But I can't articulate why, and am not sure it's better. WDYT?